src/x509_str.c

```diff
@@ -548,9 +563,31 @@ int wolfSSL_X509_verify_cert(WOLFSSL_X509_STORE_CTX* ctx)
         }
 
         depth--;
+        continue;
+
+retry:
+        /* Current certificate failed, but it is possible there is an alternative
+         * cert with the same subject key which will work.  Retry until all
+         * possible candidate certs are exhausted. */
+         WOLFSSL_MSG("X509_verify_cert current cert failed, retrying with other certs.");
+         ret = X509StoreRemoveCa(ctx->store, ctx->current_cert, WOLFSSL_TEMP_CA);
```

Comment:
It looks like this function removes a single temp CA? Is there cases where an intermediate CA unintentionally stays in the list of CA's after a failed attempt at verification and is used with the next attempt?

Reply:
Yes, this is removing the current cert from the CA list if it is present with type WOLFSSL_TEMP_CA.  Otherwise known invalid CAs will stay in the cert manager's CA list and potentially cause false verifications.
>Is there cases where an intermediate CA unintentionally stays in the list of CA's after a failed attempt at verification and is used with the next attempt?
That is correct, since the intermediate CA is potentially added to the CA list earlier with the call to X509StoreAddCa, the intermediate CA would stay in the CA list and be used for verification for all future attempts in this function despite being invalid.  This call prevents that.  Otherwise the CA isn't removed until the call to wolfSSL_CertManagerUnloadTempIntermediateCerts in exit.